File: gex/Models/Bar/ApmStats.cs
using System.Collections.Generic;

namespace gex.Models.Bar {

    public class ApmStats {

        public string GameID { get; set; } = "";

        public byte PlayerID { get; set; }

        public int ActionCount { get; set; }

        public List<ApmPeriod> Periods { get; set; } = [];

    }

    public class ApmPeriod {

        public int ActionCount { get; set; }

        public double TimeStart { get; set; }

        public double TimeDuration { get; set; }

    }

}

File: gex/src/model/ApmStats.ts

export class ApmStats {

    public gameID: string = "";
    public playerID: number = 0;
    public actionCount: number = 0;
    public periods: ApmPeriod[] = [];

    public static parse(elem: any): ApmStats {
        return {
            gameID: elem.gameID,
            playerID: elem.playerID,
            actionCount: elem.actionCount,
            periods: elem.periods.map((iter: any) => ApmPeriod.parse(iter)) 
        };
    }
}

export class ApmPeriod {
    public actionCount: number = 0;
    public timeStart: number = 0;
    public timeDuration: number = 0;

    public static parse(elem: any): ApmPeriod {
        return {
            ...elem
        };
    }
}